### DISCUSSION TITLE
VS Code Pyright extension: Feature contributions

### DISCUSSION THREAD

**Discussion by thernstig** [microsoft/pyright#6277] [General]
The VS Code Pyright extension adds several settings to VS Code:

![image](https://github.com/microsoft/pyright/assets/30827238/477b0073-9092-4853-a734-5ec98d74e750)

Coming from:

https://github.com/microsoft/pyright/blob/4c49f518a37bae49f511f853ad4ded07d266debc/packages/vscode-pyright/package.json#L118-L127

But checking the default settings.json it shows:

```jsonc
  // Offer auto-import completions.
  "python.analysis.autoImportCompletions": false,
```

Is there something awry here? The extension says the default is `true` whilst the default config says `false`.

**Comment by erictraut** [comment]
This is a setting whose default differs between the pyright extension and the pylance extension. A subset of pylance users indicated that they didn't like the auto-import feature, so the pylance team decided to disable it by default. This decision was controversial. In pyright, it's enabled by default. 

I'm not sure where you're seeing "the default settings.json" set to false. Is it possible that you have both pylance and pyright installed at the same time? You should stick with one or the other, not both.

**Reply by thernstig** [reply]
Got it. The confusion stems from that both extensions adds a top-level `python.` part to the config, meaning it is not easy to know which extension is adding what. If it were `pyright.` it would have made more sense.
